cpg_workflows/stages/gatk_sv/gatk_sv_multisample_2.py

```diff
@@ -531,18 +520,53 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
         return self.make_outputs(cohort, data=expected_d, jobs=jobs)
 
 
-def _sv_annotated_meta(output_path: str) -> dict[str, Any]:
+@stage(
+    required_stages=FilterGenotypes,
+    analysis_type='sv',
+    analysis_keys=['fresh_id_vcf'],
+    update_analysis_meta=lambda x: {'remove_sgids': EXCLUSION_FILE},
+)
+class SpiceUpSVIDs(CohortStage):
     """
-    Callable, add meta[type] to custom analysis object
+    Overwrites the GATK-SV assigned IDs with a meaningful ID
+    This new ID is based on the call attributes itself
     """
-    return {'type': 'gatk-sv-annotated', 'remove_sgids': EXCLUSION_FILE}
+
+    def expected_outputs(self, cohort: Cohort) -> dict[str, Path]:
+        return {
+            'new_id_vcf': self.prefix / 'fresh_ids.vcf.bgz',
+            'new_id_index': self.prefix / 'fresh_ids.vcf.bgz.tbi',
+        }
+
+    def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput:
+        # read the filtered VCF into the batch
+        input_vcf = inputs.as_dict(cohort, FilterGenotypes)['filtered_vcf']
+        expected_output = self.expected_outputs(cohort)
+        new_vcf = str(expected_output['new_id_vcf']).removesuffix('.vcf.bgz')
+
+        # update the IDs using a PythonJob
+        pyjob = get_batch().new_python_job('rename_sv_ids')
+        pyjob.storage('10Gi')
+        pyjob.call(rename_sv_ids, input_vcf, pyjob.output)
+
+        # then compress & run tabix on that plain text result
+        bcftools_job = get_batch().new_job('bgzip and tabix')
+        bcftools_job.image(image_path('bcftools'))
+        bcftools_job.declare_resource_group(output={'vcf.bgz': '{root}.vcf.bgz', 'vcf.bgz.tbi': '{root}.vcf.bgz.tbi'})
+        bcftools_job.command(f'bgzip -c {pyjob.output} > {bcftools_job.output["vcf.bgz"]}')  # type: ignore
+        bcftools_job.command(f'tabix {bcftools_job.output["vcf.bgz"]}')  # type: ignore
+
+        # get the output root to write to
+        get_batch().write_output(bcftools_job.output, new_vcf)
+
+        return self.make_outputs(cohort, data=expected_output, jobs=[pyjob, bcftools_job])
 
 
 @stage(
-    required_stages=FilterGenotypes,
+    required_stages=SpiceUpSVIDs,
     analysis_type='sv',
     analysis_keys=['annotated_vcf'],
-    update_analysis_meta=_sv_annotated_meta,
+    update_analysis_meta=lambda x: {'remove_sgids': EXCLUSION_FILE},
```

Comment:
I never think of using a lambda! So much better.

Reply:
I really hope this works 😅